### DISCUSSION TITLE
About DepthWiseConv2D support

### DISCUSSION THREAD

**Discussion by AnouarITI** [fastmachinelearning/hls4ml#685] [Q&A]
Hello
I wanted to load Keras's pre-trained model namely the MobileNet variants and convert them to hls. However, I got the error message that DepthWiseConv2D layers are not supported. I thought this issue was already resolved earlier. I am using the latest version v0.6.0 of hls4ml.
Any solution or hint, please?

**Comment by vloncar** [comment]
`DepthwiseConv2D` should be supported, can you provide a log? Also, I should tell you you will not succeed with synthesis of MobileNet, it is too big.

**Reply by AnouarITI** [reply]
here is some of the code:
```
MNetv1 = mobilenet.MobileNet()
MNetv1.summary()
import hls4ml

hls4ml.model.optimizer.get_optimizer('output_rounding_saturation_mode').configure(layers=['Activation'])
hls4ml.model.optimizer.get_optimizer('output_rounding_saturation_mode').configure(rounding_mode='AP_RND')
hls4ml.model.optimizer.get_optimizer('output_rounding_saturation_mode').configure(saturation_mode='AP_SAT')

MNetv1_config = hls4ml.utils.config_from_keras_model(MNetv1, granularity='model')
```


Here is the log :
```
Exception                                 Traceback (most recent call last)
Cell In [32], line 1
----> 1 MNetv1_config = hls4ml.utils.config_from_keras_model(MNetv1, granularity='model')

File ~/.local/lib/python3.8/site-packages/hls4ml/utils/config.py:139, in config_from_keras_model(model, granularity, default_precision, default_reuse_factor)
    137 for keras_layer in keras_layer_config:
    138     if keras_layer['class_name'] not in supported_layers:
--> 139         raise Exception('ERROR: Unsupported layer type: {}'.format(keras_layer['class_name']))
    140     if keras_layer['class_name'] in skip_layers:
    141         continue

Exception: ERROR: Unsupported layer type: DepthwiseConv2D
```

**Reply by vloncar** [reply]
Ah, that's in the config function. That function performs very poorly and we plan on replacing it with a more feature-rich implementation that exposes all configurable parameters of a given layer, not just the precision. I guess you can avoid using it for now. The conversion step will work, I checked that.

**Reply by AnouarITI** [reply]
Do you mean I should write the configuration by myself? Or, there is another function that generates a configuration?
